src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+
+            if ((this->unk29E == 2) && (this->unk308 != 0)) {
+                pos.y += 10.0f;
+            }
+
+            if (this->unk308 == 0) {
+                featherScale = Rand_ZeroFloat(2.0f) + 2.0f;
+            }
+            spB8.x = randPlusMinusPoint5Scaled(3.0f);
+            spB8.y = (Rand_ZeroFloat(2.0f) * 0.5f) + 2.0f;
+            spB8.z = randPlusMinusPoint5Scaled(3.0f);
+            spAC.z = spAC.x = 0.0f;
+            spAC.y = -0.15000000596f;
+
+            EnNiw_SpawnFeather(this, &pos, &spB8, &spAC, featherScale);
+        }
+        this->unk29E = 0;
+    }
+
+    EnNiw_UpdateFeather(this, globalCtx);
+
+    DECR(this->unkTimer24C);
+    DECR(this->unkTimer24E);
+    DECR(this->unkTimer250);
+    DECR(this->unkTimer252);
+    DECR(this->unkTimer254);
+    DECR(this->sfxTimer1);
+    DECR(this->flutterSfxTimer);
+    DECR(this->unusedTimer25A);
+    DECR(this->yawTimer);
+    DECR(this->unusedTimer25E);
+    DECR(this->unkTimer260);
+
+    this->actor.shape.rot = this->actor.world.rot;
+    this->actor.shape.shadowScale = 15.0f;
+    this->actionFunc(this, globalCtx);
+    Actor_SetHeight(&this->actor, this->unk308);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 60.0f, 0x1F);
+
+    if ((this->actor.floorHeight <= BGCHECK_Y_MIN) || (this->actor.floorHeight >= 32000.0f)) {
+        // if cucco is off the map?
+        Vec3f camera;
+        camera.x = globalCtx->view.focalPoint.x - globalCtx->view.eye.x;
+        camera.y = globalCtx->view.focalPoint.y - globalCtx->view.eye.y;
+        camera.z = globalCtx->view.focalPoint.z - globalCtx->view.eye.z;
+        camResult = camera.y / sqrtf(SQXYZ(camera));
+
+        this->actor.world.pos.x = this->actor.home.pos.x;
+        this->actor.world.pos.z = this->actor.home.pos.z;
+        this->actor.world.pos.y = (this->actor.home.pos.y + globalCtx->view.eye.y) + (camResult * 160.0f);
+
+        if (this->actor.world.pos.y < this->actor.home.pos.y) {
+            this->actor.world.pos.y = this->actor.home.pos.y + 300.0f;
+        }
+
+        this->actor.speedXZ = 0.0f;
+        this->actor.gravity = -2.0f;
+        Math_Vec3f_Copy(&this->unk2A4, &this->actor.home.pos);
+        Math_Vec3f_Copy(&this->unk2B0, &this->actor.home.pos);
+
+        this->unk304 = 0.0f;
+        this->unk300 = 0.0f;
+        this->unusedFloat2FC = 0.0f;
+        this->unusedFloat2F8 = 0.0f;
+        this->unusedFloat2F4 = 0.0f;
+        this->limbBRotx = 0.0f;
+        this->limbBRoty = 0.0f;
+        this->limbBRotz = 0.0f;
+        this->limb7Rotx = 0.0f;
+        this->limb7Roty = 0.0f;
+        this->limb7Rotz = 0.0f;
+        this->limbDRot = 0.0f;
+        this->limbFRot = 0.0f;
+
+        // clang-format off
+        this->niwStormActive = this->unusedCounter28C = this->unk292 = this->unk29E = this->unk298 = this->unk29A = this->unk29C = 0;
+        // clang-format on
+
+        for (i = 0; i < 0xA; ++i) {
```

Comment:
```suggestion
        for (i = 0; i < 10; i++) {
```
Only a suggestion, you don't have to take it